File: 01TwoSum/twoSum.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        int remainder = 0;
        map<int, int> needed;
        for (int i = 0; i < nums.size(); i++)
        {
            remainder = target - nums[i];
            if(needed.find(remainder) != needed.end()){
                return {needed[remainder], i};
            }
            needed[nums[i]] = i;
        }

        return {};
    }
};

int main()
{
    Solution sol;
    vector<int> input = {2, 7, 11, 15};
    vector<int> result = sol.twoSum(input, 9);
    for (int i : result)
    {
        cout << i << " ";
    }

    return 0;
};

File: 01TwoSum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let remainder = 0
    let result = {}
    for(let i = 0; i <= nums.length; i++){
        remainder = target - nums[i]
        if(remainder in result){
            return [i, result[remainder]]
        }
        result[nums[i]] = i
    }
};

console.log(twoSum([2,7,11,15], 9))